src/main/scala/beam/agentsim/agents/choice/mode/ModeChoiceMultinomialLogit.scala

```diff
@@ -46,24 +49,44 @@ class ModeChoiceMultinomialLogit(
     destinationActivity: Option[Activity],
     person: Option[Person] = None
   ): Option[EmbodiedBeamTrip] = {
+
+    def findBestIn(group: IndexedSeq[ModeCostTimeTransfer]): ModeCostTimeTransfer = {
+      if (group.size == 1) {
+        group.head
+      } else if (group.head.embodiedBeamTrip.tripClassifier.isTransit) {
+        val inputData = group
+          .map(
+            mct => mct.embodiedBeamTrip -> attributes(timeAndCost(mct), mct.transitOccupancyLevel, mct.numTransfers)
+          )
+          .toMap
+        val alternativesWithUtility = model.calcAlternativesWithUtility(inputData)
+        val chosenModeOpt = model.sampleAlternative(alternativesWithUtility, random)
```

Comment:
I think this is a good solution for now. We should make a note that eventually we should grab `model.getExpectedMaximumUtility(inputData)` from here and treat that as the utility of the chosen option

Reply:
Created issue #2863